x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -737,6 +738,25 @@ public PlanFactory visitRerankCommand(EsqlBaseParser.RerankCommandContext ctx) {
         return p -> new Rerank(source, p, inferenceId(ctx.inferenceId), queryText, visitFields(ctx.fields()));
     }
 
+    @Override
+    public PlanFactory visitCompletionCommand(EsqlBaseParser.CompletionCommandContext ctx) {
+
+        Source source = source(ctx);
+
+        if (false == EsqlCapabilities.Cap.COMPLETION.isEnabled()) {
+            throw new ParsingException(source, "COMPLETION is in preview and only available in SNAPSHOT build");
+        }
+
+        Expression prompt = expression(ctx.prompt);
+        Literal inferenceId = inferenceId(ctx.inferenceId);
+        String targetFieldName = ctx.targetField == null
+            ? Completion.DEFAULT_OUTPUT_FIELD_NAME
+            : visitQualifiedName(ctx.targetField).name();
+        Attribute targetField = new ReferenceAttribute(source, targetFieldName, DataType.TEXT);
```

Comment:
If you take a look at other commands like `Enrich`, `Keep`, `LOOKUP_JOIN` they use `NamedExpression` and use `visitQualifiedNamePattern`. Then we sometimes check the return type, because depending on the command using `*` as the field name might not be allowed.

I am not sure we should use `ReferenceAttribute` here - at most we should `UnresolvedAttribute` and then resolve it in the `Analyzer`.

Reply:
I updated the code to use an `UnresolvedAttribute` instead and will do the actual resolution in the analyzer.

Regarding the usage of `visitQualifiedNamePattern`, I do not really need to resolve pattern here but a single field. 
I can see other command that generate new fields (`MV_EXPAND` or `EVAL`) are using `visitQualifiedName`. What advantage do you expect from using `visitQualifiedNamePattern`?

Reply:
Yes, `visitQualifiedName ` return an `UnresolvedAttribute ` that can be used in the analyzer.